File: src/main/java/io/ryecrow/joe/internal/DatabaseType.java
package io.ryecrow.joe.internal;

/**
 * Type of a dbx file
 *
 * @author ryecrow
 * @since 1.0
 */
public enum DatabaseType {

    MESSAGE_DATABASE(0xc5fd746f),
    FOLDER_DATABASE(0xc6fd746f),
    OFFLINE(0x309dfe26);

    private final int uid;

    DatabaseType(int uid) {
        this.uid = uid;
    }

    public static DatabaseType getByUid(int uid) {
        for (DatabaseType ft : values()) {
            if (ft.getUid() == uid) {
                return ft;
            }
        }
        return null;
    }

    public int getUid() {
        return uid;
    }
}

File: src/main/kotlin/io/ryecrow/joe/internal/DatabaseType.kt
package io.ryecrow.joe.internal

/**
 * Type of a dbx file
 *
 * @author ryecrow
 * @since 1.0
 */
enum class DatabaseType(val uid: Int) {

    UNKNOWN(0),
    MESSAGE_DATABASE(-0x3a028b91),
    FOLDER_DATABASE(-0x39028b91),
    OFFLINE(0x309dfe26);
}

fun getDatabaseTypeByUid(uid: Int): DatabaseType {
    for (ft in DatabaseType.values()) {
        if (ft.uid == uid) {
            return ft
        }
    }
    return DatabaseType.UNKNOWN
}

